docs/DeveloperGuide.md

```diff
@@ -133,6 +133,41 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### UI Commands [Coming soon in v1.3]
+
+The mechanism to issue commands to change some aspect of the `Ui` (e.g. displaying a new panel) is facilitated by `UiCommand` abstract class. This mechanism is similar to the command pattern. 
+
+The client creates a concrete `UiCommand` that encapsulates all information needed to make a change to the Ui. Each concrete `UiCommand` implements the `UiCommand#execute(MainWindow)` method, which calls the appropriate method(s) in `MainWindow` to make a change to the UI.
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+Step 1. The user issues executes the command `viewP 1`, which displays a panel containing information about the first project in the project list. 
+
+Step 2. A `Command` is executed and a `CommandResult` object is created (see section on [Logic Component](#logic-component)) containing a `ViewProjectUiCommand` object, a concrete implementation of `UiCommand`.
+
+Step 3. The `CommandResult` is passed to the `Ui`, which gets the `UiCommand` by calling `CommandResult#getUiCommand()`.
```

Comment:
Maybe the UiCommand can be specified to be ViewProjectUiCommand since that is the current context?